src/lib/operators/case_operator.cpp

```diff
@@ -0,0 +1,226 @@
+      const auto stream_case_result = [&](const auto& case_result) {
+        if (case_result.type() == typeid(Null)) {
+          stream << "NULL";
+        } else if (case_result.type() == typeid(ColumnID)) {
+          stream << "#Col" << boost::get<ColumnID>(case_result);
+        } else if (case_result.type() == typeid(ResultDataType)) {
+          stream << "'" << boost::get<ResultDataType>(case_result) << "'";
+        }
+      };
+
+      const auto& expression = static_cast<const PhysicalCaseExpression<ResultDataType>&>(*abstract_expression);
+
+      for (const auto& clause : expression.clauses) {
+        stream << "WHEN #Col" << clause.when << " THEN ";
+        stream_case_result(clause.then);
+        stream << ", ";
+      }
+
+      stream << "ELSE ";
+      stream_case_result(expression.else_);
+    });
+
+    stream << "]";
+
+    if (expression_idx + 1 < _case_expressions.size()) stream << next_expression_separator;
+  }
+
+  stream << "}";
+
+  return stream.str();
+}
+
+std::shared_ptr<const Table> CaseOperator::_on_execute() {
+  /**
+   * Create ColumnDefinitions for the output table - consisting of the columns from the input, plus one for every
+   * _case_expression.
+   */
+  auto column_definitions = input_table_left()->column_definitions();
+  for (const auto& case_expression : _case_expressions) {
+    column_definitions.emplace_back("case_expr", case_expression->result_data_type, true);
+  }
+
+  const auto output_table = std::make_shared<Table>(column_definitions, TableType::Data);
+
+  /**
+   * For every Chunk in the input Table, create a Chunk in the output Table that contains the same columns as the input
+   * Table with the results from the CaseExpressions each as an additional Column
+   */
+  for (ChunkID chunk_id{0}; chunk_id < input_table_left()->chunk_count(); ++chunk_id) {
+    const auto input_chunk = input_table_left()->get_chunk(chunk_id);
+
+    /**
+     * Case appends a new column for every CaseExpression. All input columns are forwarded, and as Case produces a
+     * materialized Data Table, we re-use the columns from the input chunk if they are Data Columns and materialize them
+     * if they are Reference Columns.
+     */
+    ChunkColumns output_columns;
+    if (input_table_left()->type() == TableType::References) {
+      for (const auto& column : input_chunk->columns()) {
+        output_columns.emplace_back(materialize_as_value_column(*column));
+      }
+    } else {
+      output_columns = input_chunk->columns();
+    }
+
+    /**
+     * Create an output_column from each CaseExpression
+     */
+
+    // If a column in the input chunk is used as a WHEN column, it is materialized into this cache
+    std::vector<std::optional<std::vector<int32_t>>> materialized_when_columns(input_table_left()->column_count());
+
+    for (const auto& abstract_case_expression : _case_expressions) {
+      resolve_data_type(abstract_case_expression->result_data_type, [&](auto type) {
```

Comment:
here it would be good to write `[&](auto result_data_type_c)` or since we said `DataT`, maybe it should be `result_data_t`?

Reply:
Waaah, I wasn’t even sure what to think when I first looked at the naming issue. Now that you are suggesting `data_type_c`, I am even more lost and I am afraid that others will be too. :/ This is not the right place to discuss that though.